shared/modules/selectors/feature-flags.ts

```diff
@@ -0,0 +1,35 @@
+import { getCurrentChainId } from '../../../ui/selectors/selectors'; // TODO: Migrate shared selectors to this file.
+import { getNetworkNameByChainId } from '../feature-flags';
+
+type FeatureFlagsMetaMaskState = {
+  metamask: {
+    swapsState: {
+      swapsFeatureFlags: {
+        [key: string]: {
+          extensionActive: boolean;
+          mobileActive: boolean;
+          smartTransactions: {
+            expectedDeadline?: number;
+            maxDeadline?: number;
+            returnTxHashAsap?: boolean;
+          };
+        };
+      };
+    };
+  };
+};
+
+export function getFeatureFlagsByChainId(state: FeatureFlagsMetaMaskState) {
+  const chainId = getCurrentChainId(state);
+  const networkName = getNetworkNameByChainId(chainId);
+  const featureFlags = state.metamask.swapsState?.swapsFeatureFlags;
+  if (!featureFlags?.[networkName]) {
+    return null;
+  }
+  return {
+    smartTransactions: {
+      ...featureFlags[networkName].smartTransactions,
+      ...featureFlags.smartTransactions,
```

Comment:
Do we want to flip the order of these two so that the chain-specific `smartTransactions` properties would be chosen over the more general properties?